docs/HowToGuides.md

```diff
@@ -1,5 +1,63 @@
 # How To ...
 
+### Create a multi-step PR
+
+Big PRs (i.e. PRs touching more than 100 lines) are hard to review. 
+Using the _multi-step PR_ format explained below allows the review to be done incrementally.
+
+1. Dev creates the PR branch consisting of a series of commits corresponding to a sequence of logical steps, 
+   each step advancing the code base a step towards the end goal of the PR. The build should pass at each step.
+   <details>
+   <summary>Instructions</summary>
+   
+   * There Ideal approach for creating a multi-step commit sequence: 
+     * Plan the PR as a sequence of logical steps.
+     * Implement each step, committing each step as a separate commit. 
+       * Write detailed commit messages as if you were explaining your commit to a future developer.
+       
+        > Writing detailed commit messages is especially important for multi-step PRs because the commits will be 
+        > preserved after merging (i.e. the branch will be merged using a merge commit instead of 'squash and merge')
+        
+       * You may have to `squash` trivial commits or reorder commits to make the steps more logical. 
+   * Alternative approach:
```

Comment:
This is actually what usually happens in practice. We may wish to note that here, in case readers try to force themselves to follow the "ideal approach".

But thinking about it, the "ideal approach" isn't really an "approach" in the first place, and we should not label it as such. Rather, it is instead the "desired end-result" of a multi-step PR.

The thing is, the thought process which we use to write code is likely not to be the same as the ordering of commits which fulfills the requirements of a multi-step PR. (e.g. bisectabilty). I like to think of it as writing a book; readers like to read a book sequentially, starting from the introduction and ending at the conclusion. However, authors are instead encouraged to write the bulk of the story first *before* writing the introduction in order to write a cohesive story.

So, overall, I think that we should not try to word this whole section as a "follow these steps 1-2-3 to get a multi-step PR!" kind of thing. Instead, I think we should

1. First explain what a good multi-step PR is, giving good examples and bad examples, to let readers understand what the desired end-result is. This is written in prose at the start of this section.

2. *Then* explain possible approaches which readers can try out to get this desired end-result, without branding any of them as "more ideal" or "more desirable", and making it clear that it is all about *personal preference*, and readers should try them out to see what they like best. This can use the 1-2-3 numbering format. From the point of view of reviewers, we don't care *how* contributors make their PR (how the sausage is made), as long as the end result is what we want.

Maybe we can also get some ideas from https://sethrobertson.github.io/GitBestPractices/#sausage